src/main/java/org/sunbird/bpreports/service/BPReportConsumer.java

```diff
@@ -0,0 +1,457 @@
+                userInfo.put(Constants.MOBILE, personalDetails.get(Constants.MOBILE));
+                userInfo.put(Constants.GENDER, personalDetails.get(Constants.GENDER));
+                userInfo.put(Constants.DOB, personalDetails.get(Constants.DOB));
+                userInfo.put(Constants.DOMICILE_MEDIUM, personalDetails.get(Constants.DOMICILE_MEDIUM));
+                userInfo.put(Constants.CATEGORY, personalDetails.get(Constants.CATEGORY));
+            }
+
+            List<Map<String, Object>> professionalDetails = (List<Map<String, Object>>) profileDetails.get(Constants.PROFESSIONAL_DETAILS);
+            if (!CollectionUtils.isEmpty(professionalDetails)) {
+                Map<String, Object> professionalDetailsObj = professionalDetails.get(0);
+                userInfo.put(Constants.GROUP, professionalDetailsObj.get(Constants.GROUP));
+                userInfo.put(Constants.DESIGNATION, professionalDetailsObj.get(Constants.DESIGNATION));
+                userInfo.put(Constants.DOR, professionalDetailsObj.get(Constants.DOR));
+            }
+
+            Map<String, Object> employmentDetails = (Map<String, Object>) profileDetails.get(Constants.EMPLOYMENT_DETAILS);
+            if (MapUtils.isNotEmpty(employmentDetails)) {
+                userInfo.put(Constants.DEPARTMENTNAME, employmentDetails.get(Constants.DEPARTMENTNAME));
+                userInfo.put(Constants.EMPLOYEE_CODE, employmentDetails.get(Constants.EMPLOYEE_CODE));
+                userInfo.put(Constants.PINCODE, employmentDetails.get(Constants.PINCODE));
+            }
+
+            Map<String, Object> additionalProperties = (Map<String, Object>) profileDetails.get(Constants.ADDITIONAL_PROPERTIES);
+            if (MapUtils.isNotEmpty(additionalProperties)) {
+                userInfo.put(Constants.EXTERNAL_SYSTEM_ID, additionalProperties.get(Constants.ADDITIONAL_PROPERTIES));
+            }
+
+            Map<String, Object> cadreDetails = (Map<String, Object>) profileDetails.get(Constants.CADRE_DETAILS);
+            if (MapUtils.isNotEmpty(cadreDetails)) {
+                userInfo.put(Constants.CIVIL_SERVICE_TYPE, cadreDetails.get(Constants.CIVIL_SERVICE_TYPE));
+                userInfo.put(Constants.CIVIL_SERVICE_NAME, cadreDetails.get(Constants.CIVIL_SERVICE_NAME));
+                userInfo.put(Constants.CADRE_NAME, cadreDetails.get(Constants.CADRE_NAME));
+            }
+
+        }
+        return userInfo;
+    }
+
+    private String getEnrolmentStatus(WfStatusEntity wfStatusEntity) {
+        String currentStatus = wfStatusEntity.getCurrentStatus();
+        if (currentStatus.equalsIgnoreCase("SEND_FOR_MDO_APPROVAL")) {
+            return "Pending with MDO";
+        } else if (currentStatus.equalsIgnoreCase("SEND_FOR_PC_APPROVAL")) {
+            return "Pending with the Program Coordinator";
+        } else if (currentStatus.equalsIgnoreCase("APPROVED")) {
+            return "APPROVED";
+        }
```

Comment:
Must use constants string... 
And always check against constants instead of variables... 
Code should be 
Constants.SEND_FOR_MDO_APPROVAL_STATUS.equalsIgnoreCase(currentStatus)